Graphics/Implicit.hs

```diff
@@ -6,205 +6,212 @@
+    P.differenceR,
+    P.implicit,
+    P.shell,
+    P.outset,
+    P.emptySpace,
+    P.fullSpace,
+    P.withRounding,
+
+    -- * 2D primitive shapes
+    P.square,
+    P.rect,
+    P.circle,
+    P.polygon,
+
+    -- * 2D operations
+    P.rotate,
+    P.pack2,
+
+    -- * 3D primitive shapes
+    P.cube,
+    P.rect3,
+    P.sphere,
+    P.cylinder,
+    P.cylinder2,
+
+    -- * 3D operations
+    P.rotate3,
+    P.rotate3V,
+    P.pack3,
+
+    -- * Extrusions into 3D
+    P.extrude,
+    P.extrudeM,
+    P.extrudeOnEdgeOf,
+    P.rotateExtrude,
+
+    -- * OpenScad support
+    E.runOpenscad,
+
+    -- * 2D exporters
+    writeSVG,
+    writePNG2,
+    writeDXF2,
+    writeSCAD2,
+    writeGCodeHacklabLaser,
+
+    -- * 3D exporters
+    writeSTL,
+    writeBinSTL,
+    writeOBJ,
+    writeTHREEJS,
+    writeSCAD3,
+    writePNG3,
+
+    -- * Linear re-exports
+    L.V2 (V2),
+    L.V3 (V3),
+  )
+where
 
 -- The primitive objects, and functions for manipulating them.
 -- MAYBEFIXME: impliment slice operation, regularPolygon and zsurface primitives.
-import Graphics.Implicit.Primitives as P (withRounding, rect, rect3, translate, scale, mirror, complement, union, intersect, difference, unionR, intersectR, differenceR, shell, extrude, extrudeM, extrudeOnEdgeOf, sphere, cube, circle, cylinder, cylinder2, square, polygon, rotateExtrude, rotate3, rotate3V, pack3, rotate, pack2, implicit, fullSpace, emptySpace, outset, Object)
 
 -- The Extended OpenScad interpreter.
-import Graphics.Implicit.ExtOpenScad as E (runOpenscad)
 
 -- typesclasses and types defining the world, or part of the world.
-import Graphics.Implicit.Definitions as W (ℝ, ℝ2, ℝ3, SymbolicObj2, SymbolicObj3, ExtrudeMScale(C1, C2, Fn))
-
+import Graphics.Implicit.Definitions as W (ExtrudeMScale (C1, C2, Fn), SymbolicObj2, SymbolicObj3, ℝ, ℝ2, ℝ3)
 -- Functions for writing files based on the result of operations on primitives.
-import qualified Graphics.Implicit.Export as Export (writeSVG, writeDXF2, writeSTL, writeBinSTL, writeOBJ, writeSCAD2, writeSCAD3, writeTHREEJS, writeGCodeHacklabLaser, writePNG)
-
-import Linear as L (V2(V2), V3(V3), Quaternion(Quaternion))
+import qualified Graphics.Implicit.Export as Export (writeBinSTL, writeDXF2, writeGCodeHacklabLaser, writeOBJ, writePNG, writeSCAD2, writeSCAD3, writeSTL, writeSVG, writeTHREEJS)
+import Graphics.Implicit.ExtOpenScad as E (runOpenscad)
+import Graphics.Implicit.Primitives as P (Object, circle, complement, cube, cylinder, cylinder2, difference, differenceR, emptySpace, extrude, extrudeM, extrudeOnEdgeOf, fullSpace, implicit, intersect, intersectR, mirror, outset, pack2, pack3, polygon, rect, rect3, rotate, rotate3, rotate3V, rotateExtrude, scale, shell, sphere, square, translate, union, unionR, withRounding)
+import Linear as L (Quaternion (Quaternion), V2 (V2), V3 (V3))
+import Prelude (FilePath, IO)
 
 -- We want Export to be a bit less polymorphic
 -- (so that types will collapse nicely)
 
-writeSVG
```

Comment:
yeah, this same comment everywhere is a bad code smell.

Reply:
I don't think it's too bad.  The only solution I see is to have the full comment in one place and reference it in the others ("for details see analogous argument in `writeSVG`", or something).

I would just leave it as is.